AzureSQLMaintenance.txt

```diff
@@ -243,7 +243,8 @@ begin
 		and (alloc_unit_type_desc = 'IN_ROW_DATA' /*avoid LOB_DATA or ROW_OVERFLOW_DATA*/ or alloc_unit_type_desc is null /*for ColumnStore indexes*/)
 		and OBJECT_SCHEMA_NAME(idxs.object_id) != 'sys'
 		and idxs.is_disabled=0
-		and obj.type_desc != 'TF' /* Ignore table value functions */
+		and obj.type <> 'TF' /* Ignore table value functions */
+		and not exists ( select * from sys.external_tables as et where et.object_id = obj.object_id )
```

Comment:
If the [docs](https://learn.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-objects-transact-sql?view=sql-server-ver16) can be trusted it looks like external tables should have `type = 'ET'`?
```suggestion
		and obj.type not in ('ET'/*external table*/, 'TF'/*table-valued function*/)
```

Reply:
The docs cannot be trusted. That was my first approach.

Reply:
```
select
	obj.[name],
	obj.[type],
	obj.[type_desc]
from
	sys.external_tables as et
	inner join
		sys.objects as obj
	on
		obj.[object_id] = et.[object_id]
go
```

| name | type | type_desc |
|--------|--------|--------|
| v_matter| U | USER_TABLE |
| v_client | U | USER_TABLE |
| v_e_superc | U | USER_TABLE|

Reply:
Thanks for confirming.